### ISSUE TITLE
[FEATURE] Namespace all command line environment variables with `WASMCLOUD_`

### ISSUE THREAD

**Issue by brooksmtownsend** [wasmCloud/wasmCloud#1102]
**Is your feature request related to a problem? Please describe.**
Most of our flags are prefixed with `WASMCLOUD_` but there are a few that are not.
```
      --nats-host <NATS_HOST>
          NATS server host to connect to [env: NATS_HOST=] [default: 127.0.0.1]
      --nats-port <NATS_PORT>
          NATS server port to connect to [env: NATS_PORT=] [default: 4222]
      --nats-jwt <NATS_JWT>
          A user JWT to use to authenticate to NATS [env: NATS_JWT=]
      --nats-seed <NATS_SEED>
          A seed nkey to use to authenticate to NATS [env: NATS_SEED=]
      --oci-registry <OCI_REGISTRY>
          Used in tandem with `oci_user` and `oci_password` to override credentials for a specific OCI registry [env: OCI_REGISTRY=]
      --oci-user <OCI_USER>
          Username for the OCI registry specified by `oci_registry` [env: OCI_REGISTRY_USER=]
      --oci-password <OCI_PASSWORD>
          Password for the OCI registry specified by `oci_registry` [env: OCI_REGISTRY_PASSWORD=]

```

**Describe the solution you'd like**
Ideally all of these environment variables are prefixed with `WASMCLOUD_` and serde aliases are added for the previous environment variables to preserve backwards compatibility.

**Describe alternatives you've considered**
Worth mentioning that the OTEL environment variables are not namespaced as per convention, we match the standard OTEL variable syntax. Future variables like this don't have to be namespaced.

**Comment by hailelagi** [comment]
hi @brooksmtownsend would love to contribute this, is this behaviour still desired? Is my assumption correct that the desired cli ux for the env variables parsed by clap would all be prefixed for each variable above -- for example `NATS_HOST` becomes `WASMCLOUD_NATS_HOST` like so:

```rust
/// NATS server host to connect to
#[clap(long = "nats-host", env = "WASMCLOUD_NATS_HOST", alias="NATS_HOST")]
pub nats_host: Option<String>,
```

**Comment by thomastaylor312** [comment]
Hey @hailelagi! So the only thing that needs to change is just the environment variable. The actual field name can remain the same. 

> and lastly some of the env flags such as OCI_USER, OCI_PASSWORD, OCI_REGISTRY, NATS_JWT and NATS_SEED seem have either been renamed or removed?

I am not sure I follow here. I just checked on main and saw all of those variables present (e.g. https://github.com/wasmCloud/wasmCloud/blob/bb4d5b3e0bdd8777cf73c30413dea2aa2f068ef9/src/main.rs#L184C16-L184C28). Did I misunderstand the question?

**Comment by hailelagi** [comment]
> Did I misunderstand the question?
@thomastaylor312 nope not at all, I restricted my search to the sub dir `./wash-cli` so I didn't find it when I grepped locally.
Thank you so much, this helps a tonne!

**Comment by brooksmtownsend** [comment]
The `wash` part of this ticket was addressed by @hailelagi in #1243
